src/backend/marsha/core/tests/test_xapi.py

```diff
@@ -243,16 +263,112 @@ def test_xapi_statement_live_video(self):
                 "definition": {
                     "type": "http://id.tincanapi.com/activitytype/webinar",
                     "name": {"en-US": "test video xapi"},
-                    "extensions": {
-                        "https://w3id.org/xapi/acrossx/extensions/school": "ufr",
-                        "http://adlnet.gov/expapi/activities/course": "mathematics",
-                        "http://adlnet.gov/expapi/activities/module": "001",
-                    },
                 },
                 "id": "uuid://68333c45-4b8c-4018-a195-5d5e1706b838",
                 "objectType": "Activity",
             },
         )
+        self.assertEqual(
+            statement["context"],
+            {
+                "extensions": {
+                    "https://w3id.org/xapi/video/extensions/session-id": "a6151456-18b7-"
+                    "43b4-8452-2037fed588df"
+                },
+                "contextActivities": {
+                    "category": [{"id": "https://w3id.org/xapi/video"}],
+                    "parent": [
+                        {
+                            "id": "course-v1:ufr+mathematics+0001",
+                            "objectType": "Activity",
+                            "definition": {
+                                "type": "http://adlnet.gov/expapi/activities/course"
+                            },
+                        }
+                    ],
+                },
+            },
+        )
+        self.assertEqual(statement["verb"], base_statement["verb"])
+        self.assertEqual(statement["id"], base_statement["id"])
+        self.assertEqual(statement["result"], base_statement["result"])
+
+    def test_xapi_statement_missing_context_id(self):
```

Comment:
I don't clearly get the point of this test. If the context_id is missing, hence the parent contextActivities is not created and then we should raise an error concerning the attribute missing, shouldn't we ?

Reply:
No we should not raise an error. If the context_id is missing, the parent contextActivities is not added and the statement is sent.

Reply:
So this test assert that without `context_id` the statement is correctly build without parent contextActivities

Reply:
Alright, got it!